hack/import-restrictions.json

```diff
@@ -180,6 +180,7 @@
     ],
     "allowedImportPackageRoots": [
       "vendor/k8s.io/apimachinery",
+      "vendor/k8s.io/kubernetes/pkg/proxy/config",
```

Comment:
> I think this won't be allowed - @deads2k for his goals here

Our api packages should only be for the type definitions themselves.  We're going to build a client-go library for openshift (see attempts here: https://github.com/openshift/client-go/pull/2) and the dependencies for that library need to be reasonable.  Our API types are required, but the interfaces which make them more useable for runtime purposes aren't generally needed by the client, so they should live a separate package.